### PR TITLE
ocm: enhance invitation flow

### PR DESCRIPTION
**Pull request by labkode**
Closes https://github.com/sciencemesh/sciencemesh/issues/22

### CODE REVIEW

File: spec.yaml [hunk 1]
```diff
@@ -355,42 +355,47 @@ paths:
               type: string
           schema:
             $ref: "#/definitions/Error"
-  /invite/accept:
+  invite/forward:
```
Comment (ishank011) [inline spec.yaml]: /invite/forward

File: spec.yaml [hunk 2]
```diff
@@ -355,42 +355,47 @@ paths:
...
     post:
-      summary: Accept an invite to a shared document
-      description: Foo
+      summary: Forward an invitation to start sharing
+      description: Sends an invitation to an end-user (by mail for example)
```
Comment (ishank011) [inline spec.yaml]: The forward endpoint doesn't send the mail, right? We don't have that functionality that. This endpoint is exposed via a link which would be provided to the user (and in the current case, through the phoenix app), and this just calls /invite/accept on the original user's side (as described [here](https://github.com/sciencemesh/sciencemesh/issues/9)).

**Review by ishank011 — COMMENTED** [review]
Hi Hugo. A couple of comments.

File: spec.yaml (hunk 1 above)
Comment (labkode) [reply spec.yaml]: Thanks

File: spec.yaml (hunk 2 above)
Comment (labkode) [reply spec.yaml]: @ishank011 right, I corrected the wording.

**Review by ishank011 — APPROVED** [review]

**Comment by labkode** [comment]
@LovisaLugnegard can you merge into your branch? Like that your PR will be updated automatically.

File: spec.yaml [hunk 3]
```diff
@@ -355,42 +355,47 @@ paths:
               type: string
           schema:
             $ref: "#/definitions/Error"
-  /invite/accept:
+  /invite/forward:
     post:
-      summary: Accept an invite to a shared document
-      description: Foo
+      summary: Forward an invitation to start sharing
```
Comment (LovisaLugnegard) [inline spec.yaml:360]: Is there a reason for using both "invite" and "invitation" instead of just one of them?

File: spec.yaml [hunk 4]
```diff
@@ -355,42 +355,47 @@ paths:
...
-          description: The JSON object to notify OCM that an invite has been accepted.
+          description: The JSON object representing an invitation to send to an end-user.
           required: true
           schema:
-            $ref: "#/definitions/AcceptInvite"
+            $ref: "#/definitions/Invite"
       responses:
         200:
-          description: Auth ok!
+          description: Invitation sent.
```
Comment (LovisaLugnegard) [inline spec.yaml:371]: Invitation forwarded?

File: spec.yaml [hunk 5]
```diff
@@ -355,42 +355,47 @@ paths:
...
         400:
-          description: Invalid token?
+          description: |
+            Bad request due to invalid parameters, e.g. email is not valid.
```
Comment (LovisaLugnegard) [inline spec.yaml:374]: Should probably remove "email" and maybe put "provider"?

File: spec.yaml (hunk 4 above)
Comment (labkode) [reply spec.yaml:371]: Yes, can you update it @LovisaLugnegard

File: spec.yaml [hunk 6]
```diff
@@ -355,42 +355,47 @@ paths:
...
           schema:
-            $ref: "#/definitions/Error" 
-  invite/forward:
+            $ref: "#/definitions/400"            
+  /invite/accept:
     post:
-      summary: Forward information from invite to the cloud provider.
-      description: Foo
+      summary: Accept an invititation to start sharing.
+      description: Accepts an invitation to the shared document so the provider can initiate the OCM share creation.
```
Comment (LovisaLugnegard) [inline spec.yaml:380]: I think this description needs to be updated, I'm not sure exactly how though. But isn't this an invitation to be able to share with someone ie in some sense decoupled from the sharing itself, like:
1. Invite A
2. Now one can share with A?

**Review by LovisaLugnegard — COMMENTED** [review]
Some smaller comments and thoughts.

File: spec.yaml (hunk 6 above)
Comment (labkode) [reply spec.yaml:380]: Yes, you accept the invitation to create a share with the person. Invitation and share creation are both decoupled. We can remove the shared document part.

**Pull request merged**